tools/md-formatter/form.js

```diff
@@ -0,0 +1,254 @@
+</svg>`;
+
+const style = await getStyle(import.meta.url);
+
+export function constructTable(entries) {
+  const mdTable = `
+      <table>
+        <tbody>
+          <tr>
+            <td colspan="2">
+              <p>metadata</p>
+            </td>
+          </tr>
+        </tbody>
+      </table>
+    `;
+
+  const tableObject = new DOMParser().parseFromString(mdTable, 'text/html');
+  const table = tableObject.querySelector('table');
+  const body = table.querySelector('tbody');
+
+  Object.keys(entries).forEach((key) => {
+    const tr = document.createElement('tr');
+
+    const keyTd = document.createElement('td');
+    const keyP = document.createElement('p');
+    keyP.innerText = key.toLowerCase();
+    keyTd.append(keyP);
+
+    const valueTd = document.createElement('td');
+    const valueP = document.createElement('p');
+    valueP.innerText = entries[key];
+    valueTd.append(valueP);
+
+    tr.append(keyTd, valueTd);
+    body.append(tr);
+  });
+
+  return table;
+}
+export class MdForm extends LitElement {
+  static properties = {
+    _title: { state: true },
+    _formFields: { state: true },
+    _optionsErrorMessage: { type: String },
+    _metadataOptions: { state: true },
+    _availabileFieldKeys: { type: Array },
+    _selectOptions: { type: Array },
+    _currentPropertySelect: { state: true },
+    _currentValueList: { state: true },
+    _addedFields: { type: Array },
+  };
+
+  constructor() {
+    super();
+    this._title = 'Metadata Builder';
+    this._defaultFields = [
+      { key: 'Title', value: 'page title' },
+      { key: 'Description', value: 'page description' },
+    ];
+    this._addedFields = [];
+    this._optionsErrorMessage = '';
+    this._metadataOptions = {};
+    this._availabileFieldKeys = [];
+    this._selectOptions = [];
+    this._currentPropertySelect = '';
+    this._currentValueList = [];
+    this._addedFields = [];
+  }
+
+  async getMetadataOptions() {
+    const sheetUrl = 'https://main--da-bacom--adobecom.aem.live/drafts/slavin/nobu/allowed-ppn.json';
+    const resp = await fetch(sheetUrl);
+    if (!resp.ok) {
+      this._optionsErrorMessage = 'Failed to fetch metadata options';
+      return {};
+    }
+    const json = await resp.json();
+    return json;
+  }
+
+  setSelectOptions() {
+    if (!this._metadataOptions?.data?.length) return [];
+
+    const firstItem = this._metadataOptions.data[0];
+    const keys = Object.keys(firstItem);
+    const selectOptions = keys.map((key) => ({ keyName: key, values: [] }));
+    const fieldKeys = [];
+
+    this._metadataOptions.data.forEach((item) => {
+      keys.forEach((key) => {
+        if (item[key]) {
+          selectOptions.find((option) => option.keyName === key).values.push(item[key]);
+        }
+        if (!fieldKeys.includes(key)) {
+          fieldKeys.push(key);
+        }
+      });
+    });
+
+    return [selectOptions, fieldKeys];
+  }
```

Comment:
There has to be a simpler way of doing this. Mapping and then nested forEach seems inefficient and is a bit confusing to read. Maybe there's a different way to input the data?

Reply:
Sounds good, refactored. Let me know what you think.